torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -282,8 +282,11 @@ std::shared_ptr<SugaredValue> ModuleValue::attr(
     return std::make_shared<OverloadedMethodValue>(
         self_, py::cast<std::vector<std::string>>(overloads));
   }
+  if (!py::hasattr(py_module_, field.c_str())) {
+    throw ErrorReport(loc) << "attribute '" << field << "' does not exist";
+  }
 
-  if (py::object attr = py::getattr(py_module_, field.c_str(), py::none())) {
+  if (py::object attr = py::getattr(py_module_, field.c_str())) {
```

Comment:
Code here no longer makes sense. py::getattr is always going to return true and the throw below it is no longer used.

Reply:
Also a minor nit: we should preserve the form of the error message that is below because it matches what CPython says: `module has no attribute '{}'`

Reply:
I think the code never made sense right? It would assign `py::none()` if it didn't exist, which evaluates to true.

If it worked this issue would have never been raised.